pkg/utils/rebaseTodo_test.go

```diff
@@ -0,0 +1,271 @@
+package utils
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/fsmiamoto/git-todo-parser/todo"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRebaseCommands_moveTodoDown(t *testing.T) {
+	type scenario struct {
+		testName      string
+		todos         []string
+		shaToMoveDown string
+		expectedErr   string
+		expectedTodos []string
+	}
+
+	scenarios := []scenario{
+		{
+			testName: "simple case 1 - move to beginning",
+			todos: []string{
+				"pick 1234",
+				"pick 5678",
+				"pick abcd",
+			},
+			shaToMoveDown: "5678",
+			expectedErr:   "",
+			expectedTodos: []string{
+				"pick 5678",
+				"pick 1234",
+				"pick abcd",
+			},
+		},
+		{
+			testName: "simple case 2 - move from end",
+			todos: []string{
+				"pick 1234",
+				"pick 5678",
+				"pick abcd",
+			},
+			shaToMoveDown: "abcd",
+			expectedErr:   "",
+			expectedTodos: []string{
+				"pick 1234",
+				"pick abcd",
+				"pick 5678",
+			},
+		},
+		{
+			testName: "skip an invisible todo",
+			todos: []string{
+				"pick 1234",
+				"pick abcd",
+				"label mylabel",
+				"pick 5678",
+				"pick def0",
+			},
+			shaToMoveDown: "5678",
+			expectedErr:   "",
+			expectedTodos: []string{
+				"pick 1234",
+				"pick 5678",
+				"pick abcd",
+				"label mylabel",
+				"pick def0",
+			},
+		},
+
+		// Error cases
+		{
+			testName: "commit not found",
+			todos: []string{
+				"pick 1234",
+				"pick 5678",
+				"pick abcd",
+			},
+			shaToMoveDown: "def0",
+			expectedErr:   "Todo def0 not found in git-rebase-todo",
+			expectedTodos: []string{
+				"pick 1234",
+				"pick 5678",
+				"pick abcd",
+			},
+		},
+		{
+			testName: "trying to move first commit down",
+			todos: []string{
+				"pick 1234",
+				"pick 5678",
+				"pick abcd",
+			},
+			shaToMoveDown: "1234",
+			expectedErr:   "Trying to move first commit down",
+			expectedTodos: []string{
+				"pick 1234",
+				"pick 5678",
+				"pick abcd",
+			},
+		},
+		{
+			testName: "trying to move commit down when all commits before are invisible",
+			todos: []string{
+				"label mylabel",
+				"reset otherlabel",
+				"pick 1234",
+				"pick 5678",
+			},
+			shaToMoveDown: "1234",
+			expectedErr:   "Trying to move first commit down",
+			expectedTodos: []string{
+				"label mylabel",
+				"reset otherlabel",
+				"pick 1234",
+				"pick 5678",
+			},
+		},
+	}
+
+	for _, s := range scenarios {
+		t.Run(s.testName, func(t *testing.T) {
+			fileName := filepath.Join(os.TempDir(), "git-rebase-todo-test")
+
+			err := os.WriteFile(fileName, []byte(strings.Join(s.todos, "\n")+"\n"), 0o600)
```

Comment:
I would refactor the function to take the contents of the file, and to return the updated contents of the file (or just the rearranged todos, whichever is easier), so that in these unit tests we don't actually need to hit the filesystem

Reply:
Makes sense. See e9999f3aa8f5

Reply:
I just realized that I need the functions taking a filename from another place in a future PR, so I added another fixup commit to move them back to the utils file, and rename the other ones to lowercase. See f10aa6152f56.